### PR TITLE
Add opf-observatorium loki monitoring

### LINKED ISSUE #75: Add monitoring for the loki deployment
Add service monitor for loki
Add new Prometheus deployment for opf-observatorium namespace

### PR DESCRIPTION
**Pull request by 4n4nd**
Add Prometheus and Prometheus Operator deployment  using ODH
Add ServiceMonitor for Loki
Update opf-monitoring Prometheus to monitor opf-observatorium prometheus
Update opf-monitoring metric retention to 7d from default (24H)

Closes #75

### CODE REVIEW

File: observatorium/base/instance/monitoring/prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: prometheus
+  labels:
+    prometheus: k8s
+spec:
+  replicas: 1
+  serviceAccountName: prometheus-k8s
+  securityContext: {}
+  serviceMonitorSelector:
+    matchLabels:
```
Comment (HumairAK) [inline observatorium/base/instance/monitoring/prometheus.yaml:13]: Don't you need to add this label to the service monitor?

**Comment by HumairAK** [comment]
So this looks fine to me, seems to be imitating the setup we use for how we're monitoring JH. 

I wonder how this squares with [this](https://github.com/operate-first/apps/issues/57) investigation. If it's successful then we technically don't need to be adding another prometheus instance here. On the other hand we could just add this for now and remove it later since we'll have to do it for other services like JH anyway. 

Thoughts? @anishasthana  @tumido

**Review by HumairAK — APPROVED** [review]

**Comment by 4n4nd** [comment]
> On the other hand we could just add this for now and remove it later since we'll have to do it for other services like JH anyway.

I think this is the way to go for now. I am not too optimistic about #57 tbh, but i still need to do more testing.

**Comment by anishasthana** [comment]
I'd like to confirm the results of #57  one way or the other. I'd vote for holding off on merging this until that's determined (sorry Anand :-) ).

**Comment by tumido** [comment]
I agree. I don't like seeing more than one instance of Prometheus and if there's a way to avoid it, I'd try that.

**Comment by 4n4nd** [comment]
> I agree. I don't like seeing more than one instance of Prometheus and if there's a way to avoid it, I'd try that.

Actually, I like multiple Prometheus instances. It's like poor man's "sharding". But I guess we are not at a point where we need to worry about scaling Prometheus yet.

**Comment by tumido** [comment]
You can scale and shard a single prometheus just fine. The operator itself supports it. With multiple instances of the same deployment, you would also need to keep them in sync and you need to know from which route to pull the data, compared to a single route. That's a mess.

**Comment by 4n4nd** [comment]
> The operator itself supports it

Sure, but it's an experimental feature. 
> With multiple instances of the same deployment

You still get the same issues there, each shard means a separate statefulset.

**Comment by tumido** [comment]
> You still get the same issues there, each shard means a separate statefulset.

Precisely my point why to keep a single instance until we face issues we can't solve with deployment scaling.

**Pull request closed without merging**